pkg/compactor/compactor.go

```diff
@@ -55,40 +55,38 @@ func (cp *Compactor) Compact(ctx context.Context, opts *brtypes.CompactOptions)
 	cp.logger.Info("Start compacting")
 
 	// Deepcopy restoration options ro to avoid any mutation of the passing object
-	cmpctOptions := opts.RestoreOptions.DeepCopy()
+	compactorRestoreOptions := opts.RestoreOptions.DeepCopy()
 
-	// If no basesnapshot is found, abort compaction as there would be nothing to compact
-	if cmpctOptions.BaseSnapshot == nil {
+	// If no base snapshot is found, abort compaction as there would be nothing to compact
+	if compactorRestoreOptions.BaseSnapshot == nil {
 		cp.logger.Error("No base snapshot found. Nothing is available for compaction")
 		return nil, fmt.Errorf("no base snapshot found. Nothing is available for compaction")
 	}
 
-	// Set a temporary etcd data directory for embedded etcd
-	prefix := cmpctOptions.Config.RestoreDataDir
-	if prefix == "" {
-		prefix = "/tmp"
-	}
-	cmpctDir, err := os.MkdirTemp(prefix, "compactor-")
+	cp.logger.Infof("Creating temporary etcd directory %s for restoration.", compactorRestoreOptions.Config.DataDir)
+	err := os.MkdirAll(compactorRestoreOptions.Config.DataDir, 0700)
 	if err != nil {
 		cp.logger.Errorf("Unable to create temporary etcd directory for compaction: %s", err.Error())
 		return nil, err
 	}
 
-	defer os.RemoveAll(cmpctDir)
-
-	cmpctOptions.Config.RestoreDataDir = cmpctDir
+	defer func() {
+		if err := os.RemoveAll(compactorRestoreOptions.Config.DataDir); err != nil {
+			cp.logger.Errorf("Failed to remove temporary etcd directory %s: %v", compactorRestoreOptions.Config.DataDir, err)
+		}
+	}()
 
 	// Then restore from the snapshots
 	r := restorer.NewRestorer(cp.store, cp.logger)
-	embeddedEtcd, err := r.Restore(*cmpctOptions, nil)
+	embeddedEtcd, err := r.Restore(*compactorRestoreOptions, nil)
 	if err != nil {
 		return nil, fmt.Errorf("unable to restore snapshots during compaction: %v", err)
 	}
 
-	cp.logger.Info("Restoration for compaction is over")
+	cp.logger.Info("Restoration for compaction is done.")
 	// There is a possibility that restore operation may not start an embedded ETCD.
 	if embeddedEtcd == nil {
-		embeddedEtcd, err = miscellaneous.StartEmbeddedEtcd(cp.logger, cmpctOptions)
+		embeddedEtcd, err = miscellaneous.StartEmbeddedEtcd(cp.logger, compactorRestoreOptions)
```

Comment:
This case here is not to handle any errors from `Restore`. If there's an error returned from `Restore`, `Compact` also returns error on line 83.

This here is to handle the case where no delta snapshots were found to be applied. Let's say the user forcefully runs a compaction operation on a snapstore which has only a full snapshot. In such a case, etcdbr shouldn't simply return an error that "there was nothing to compact" and also at the same time shouldn't do nothing - because compaction operation is the combination of restoraotion, etcd compaction (on the DB), defragmentation and then new full snapshot. So a compaction operation on a single full snapshot should still push a new size-optimized full snapshot to the bucket.

Hence, this extra code path to start the embedded etcd if at all it was not already started by `Restore`.

Again, all I've done in `compactor` package is change the variable names, nothing else.